src/GitTabExpansion.ps1

```diff
@@ -359,9 +359,11 @@ function GitTabExpansionInternal($lastBlock, $GitStatus = $null) {
 
         # Handles git checkout <ref>
         "^(?:checkout).* (?<ref>\S*)$" {
-            gitBranches $matches['ref'] $true
-            gitRemoteUniqueBranches $matches['ref']
-            gitTags $matches['ref']
+            $script:gitBranches = @(gitBranches $matches['ref'] $true)
```

Comment:
You aren't trying to cache the branches, so you probably shouldn't save them in a script scope variable.
I think using the pipeline like this might be a bit more natural and avoids creating multiple arrays:
````powershell
& {
    gitBranches $matches['ref'] $true
    gitRemoteUniqueBranches $matches['ref']
    gitTags $matches['ref']
} | Select-Object -Unique
````
You could use a function instead of the script block like I used here.

Reply:
Thanks for the suggestion, you're right that this is the more efficient way of doing it.